### PR TITLE
feat(manager/helmfile): Support oci urls in helmfile

### PR DESCRIPTION
**Pull request by jonasrutishauser**
## Changes
This PR add support for `oci://` urls in helmfile.

## Context
Helm uses urls of the form `oci://<host>/<path>/<chart>` which is also supported by helmfile.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/helmfile/extract.spec.ts [hunk 1]
```diff
@@ -334,10 +337,19 @@ describe('modules/manager/helmfile/extract', () => {
             depName: 'example',
             datasource: 'docker',
             packageName: 'ghcr.io/example/oci-repo/example',
+            registryUrls: ['ghcr.io/example/oci-repo'],
```
Comment (viceice) [inline lib/modules/manager/helmfile/extract.spec.ts:340]: remove, should not have a registry url, it's already embedded in packageName

File: lib/modules/manager/helmfile/extract.spec.ts [hunk 2]
```diff
@@ -334,10 +337,19 @@ describe('modules/manager/helmfile/extract', () => {
...
             currentValue: '3.3.0',
             depName: 'jenkins',
+            registryUrls: ['https://charts.jenkins.io'],
+          },
+          {
+            currentValue: '0.4.2',
+            depName: 'url-example',
+            datasource: 'docker',
+            packageName: 'ghcr.io/example/oci-repo/url-example',
+            registryUrls: ['ghcr.io/example/oci-repo'],
```
Comment (viceice) [inline lib/modules/manager/helmfile/extract.spec.ts:352]: same

File: lib/modules/manager/helmfile/extract.ts [hunk 3]
```diff
@@ -130,7 +138,11 @@ export async function extractPackageFile(
       const repository = doc.repositories?.find(
         (repo) => repo.name === repoName,
       );
-      if (repository?.oci) {
+      if (isOciUrl(dep.chart)) {
+        res.datasource = DockerDatasource.id;
+        res.packageName = repoName + '/' + depName;
+        res.registryUrls = [repoName];
```
Comment (viceice) [inline lib/modules/manager/helmfile/extract.ts:144]:
```suggestion
```

File: lib/modules/manager/helmfile/extract.ts [hunk 4]
```diff
@@ -130,7 +138,11 @@ export async function extractPackageFile(
       const repository = doc.repositories?.find(
         (repo) => repo.name === repoName,
       );
-      if (repository?.oci) {
+      if (isOciUrl(dep.chart)) {
+        res.datasource = DockerDatasource.id;
+        res.packageName = repoName + '/' + depName;
```
Comment (viceice) [inline lib/modules/manager/helmfile/extract.ts:143]:
```suggestion
        res.packageName = dep.chart;
```
it should probably use registry aliases, like below

Comment (jonasrutishauser) [reply lib/modules/manager/helmfile/extract.ts:143]: I tested against a test repository and it won't work with this suggestion, as the packageName must be the full name.

The registry aliases never make sense with an oci url, as it is already the complete url/name.

**Review by viceice — APPROVED** [review]

**Pull request merged**